src/archivist/recorder/repositories/git/dataMapper.js

```diff
@@ -0,0 +1,62 @@
+import path from 'path';
+
+import mime from 'mime';
+
+import Record from '../../record.js';
+
+mime.define({ 'text/markdown': ['md'] }, true); // ensure extension for markdown files is `.md` and not `.markdown`
+
+export const COMMIT_MESSAGE_PREFIX = {
+  startTracking: 'Start tracking',
+  refilter: 'Refilter',
+  update: 'Update',
+};
+
+export const COMMIT_MESSAGE_PREFIXES_REGEXP = new RegExp(`^(${COMMIT_MESSAGE_PREFIX.startTracking}|${COMMIT_MESSAGE_PREFIX.refilter}|${COMMIT_MESSAGE_PREFIX.update})`);
+export default class GitDataMapper {
+  static toPersistence(record, prefixMessageToSnapshotId) {
+    const { serviceId, documentType, isRefilter, snapshotId, mimeType, isFirstRecord } = record;
+
+    let prefix = isRefilter ? COMMIT_MESSAGE_PREFIX.refilter : COMMIT_MESSAGE_PREFIX.update;
+
+    prefix = isFirstRecord ? COMMIT_MESSAGE_PREFIX.startTracking : prefix;
```

Comment:
```suggestion
    let prefix = isFirstRecord ? COMMIT_MESSAGE_PREFIX.startTracking :
    isRefilter ? COMMIT_MESSAGE_PREFIX.refilter :
    COMMIT_MESSAGE_PREFIX.update;
```

Reply:
Nested ternary expression can be hard to read so we choose to disallow them with https://eslint.org/docs/rules/no-nested-ternary